src/ClapDetector.js

```diff
@@ -0,0 +1,168 @@
+// @flow
+
+import React from 'react';
+import * as FeatureDetection from './FeatureDetection';
+
+type ClapDetectorProps = {
+    onDetect: (string) => void
+}
+
+type AppContext = {
+    mediaDevicesIsAvailable: boolean,
+    webAudioAPIIsAvailable: boolean
+};
+
+type ClapDetectorState = {
+    contextAvailable: boolean,
+    audioInputDeviceIds: Array<string>
+}
+
+export default class ClapDetector extends React.Component<ClapDetectorProps, ClapDetectorState> {
+    appContext: AppContext;
+    constructor(props: ClapDetectorProps) {
+        super(props);
+
+        this.appContext = {
+            mediaDevicesIsAvailable: FeatureDetection.mediaDevicesIsAvailable(),
+            webAudioAPIIsAvailable: FeatureDetection.webAudioAPIIsAvailable()
+        };
+
+        this.state = {
+            contextAvailable: false,
+            audioInputDeviceIds: [],
+        };
+    }
+
+    componentDidMount() {
+        if (this.appContext.mediaDevicesIsAvailable) {
+            navigator.mediaDevices.enumerateDevices().then((devices) => {
+                console.log(devices);
+                let deviceIds = [];
+                for (let i=0, numDevices=devices.length; i<numDevices; i++) {
+                    if (devices[i].kind === 'audioinput') deviceIds.push(devices[i].deviceId);
+                }
+                this.setState({
+                    contextAvailable : true,
+                    audioInputDeviceIds : deviceIds
+                });
+            }, (error) => {
+                console.log(error.name);
+                console.log(error.message);
+            });
+        }
+    }
+
+    componentDidUpdate(prevProps: {}, prevState: ClapDetectorState) {
+        if (this.state.audioInputDeviceIds !== prevState.audioInputDeviceIds) {
+            if (this.state.audioInputDeviceIds.length === 0) {
+                console.log('state not updated');
+            } else {
+                const AudioContext = window.AudioContext || window.webkitAudioContext;
+                const context = new AudioContext();
+
+                Promise.all([
+                    this.addAudioStreamNode(this.state.audioInputDeviceIds[0], context),
+                    this.addAudioStreamNode(this.state.audioInputDeviceIds[1], context),
+                    this.addAudioStreamNode(this.state.audioInputDeviceIds[3], context)
+                ]).then((analysers) => {
+                    const defaultInputAnalyser = analysers[0];
+                    const externalInputAnalyser = analysers[1];
+                    const wirelessInputAnalyser = analysers[2];
+                    const scriptProcessorAnalysisNode = context.createScriptProcessor(2048,2,1);
```

Comment:
According the MDN, this function is deprecated:

https://developer.mozilla.org/en-US/docs/Web/API/BaseAudioContext/createScriptProcessor